### PR TITLE
meson: include pre-generated man pages & HTML docs in dist tarballs

### PR DESCRIPTION
**Pull request by thesamesam**
Bug: https://github.com/iputils/iputils/issues/479
Bug: https://bugs.gentoo.org/908817
Bug: https://bugs.gentoo.org/920901

### CODE REVIEW

**Comment by thesamesam** [comment]
This is a followup to https://github.com/iputils/iputils/issues/479. I'm sorry I didn't end up reviewing the work there, it's been a very hectic couple of months :(

File: meson.build [hunk 1]
```diff
@@ -218,8 +218,11 @@ if build_arping == true
 	endif
 endif
 
-if build_mans == true or build_html_mans == true
-	subdir ('doc')
+if meson.version().version_compare('>=0.58.0')
+	if build_mans == true or build_html_mans == true
+		subdir ('doc')
+		meson.add_dist_script('meson-build-dist-man.sh')
+	endif
```
Comment (eli-schwartz) [inline meson.build:225]: Typo for 0.48 (when dist scripts were added)?

Anyway I think you don't want to skip the entire doc/ directory on too old versions of meson.

**Review by eli-schwartz — CHANGES_REQUESTED** [review]

File: meson.build (hunk 1 above)
Comment (thesamesam) [reply meson.build:225]: Yeah, I think I was copying the 0.58.0 from `doc`. Good catch on the skipping...

**Comment by pevik** [comment]
@nmeyerhans any opinion about this?

I would really prefer people either download build man page dependencies or disable building it. But understand the use case.
The implementation needs minor cleanup (script in `tools/` etc.), but that's a minor detail (I can fix these tiny issues if we agree we want this approach).

**Comment by thesamesam** [comment]
FWIW, I wouldn't ask if it weren't a core system package, and I'm also happy to maintain and handle any problems with this side of things if they crop up. But I also understand if you want to leave it too. Thank you for considering it either way!

File: meson-build-dist-man.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# This script should be invoked by meson itself (via 'meson dist')
+# See https://github.com/mesonbuild/meson/issues/2166 and more specifically,
+# https://github.com/mesonbuild/meson/issues/2166#issuecomment-629696911.
+set -eu
+
+cd "${MESON_DIST_ROOT}"
```
Comment (pevik) [inline meson-build-dist-man.sh:7]: nit: could you please change to   `cd "$MESON_DIST_ROOT"`

File: meson-build-dist-man.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+mkdir build
```
Comment (pevik) [inline meson-build-dist-man.sh:8]: I wonder if we should use `mktemp`, that would avoid to clash (name is too generic).

File: meson-build-dist-man.sh (hunk 2 above)
Comment (pevik) [reply meson-build-dist-man.sh:7]: Also, please move the script to the `tools/` directory.

**Comment by pevik** [comment]
@thesamesam I understand all distros would benefit from it (less dependencies shorten build + require less resources), so I'm for this (I'm sorry I haven't managed to handle it before the release). Could you please rebase your pull request and handle minor issues I reported?

**Comment by pevik** [comment]
@thesamesam FYI I have ready slightly modified branch https://github.com/pevik/iputils/tree/meson-dist.fixes, I just wait if we get any input from https://github.com/mesonbuild/meson/issues/2166#issuecomment-1938765316 to get this running on cross compilation, or if I need to disable it with help of [`meson.is_cross_build()`](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonis_cross_build) as you suggested.

**Comment by pevik** [comment]
I wonder why `FAILED unittests/linuxliketests.py::LinuxlikeTests::test_generate_gir_with_address_sanitizer` failed - it timed out due:

```
stdout_seq = [b"ninja explain: deps for 'resources/simple-resources-test.p/meson-generated_.._simple-resources.c.o' are missing\n",...ces.c.o is dirty\n', b"ninja explain: deps for 'resources/simple-resources-test.p/simple-main.c.o' are missing\n", ...]

```

**Comment by thesamesam** [comment]
Thank you!

**Comment by pabs3** [comment]
FTR, other distros like Debian strongly prefer to not have generated files in the VCS and tarballs. Instead everything should always be built from source. More info on the Debian [AutoGeneratedFiles](https://wiki.debian.org/AutoGeneratedFiles) wiki page.

**Comment by thesamesam** [comment]
You're free to not use them in that case.

**Comment by nmeyerhans** [comment]
> FTR, other distros like Debian strongly prefer to not have generated files in the VCS and tarballs. Instead everything should always be built from source. More info on the Debian [AutoGeneratedFiles](https://wiki.debian.org/AutoGeneratedFiles) wiki page.

Debian doesn't use the upstream release tarballs at all, but builds from the git tags directly. So none of this is relevant here.

**Pull request closed without merging**